src/core/binary/node/decoder.ts

```diff
@@ -21,7 +23,16 @@ export async function generateTextualPlist(
     );
   } else {
     const content = await decodeBinaryPlist(document.uri);
-    const plistContent = plist.build(content as plist.PlistValue);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const plistContent = plist.build(content as plist.PlistValue, {}, {
+      invalidCharReplacement: '�',
+    } as CreateOptions);
+    if (plistContent.indexOf('�') !== -1) {
+      vscode.window.showWarningMessage(
+        'This file contains “�” characters, likely due to control characters in strings. If you save the file, these replacement characters will remain.'
```

Comment:
I like this solution, but I think the warning message could be confusing. A user receiving this message is likely to through their file to try and find the `�` character that "this file contains", only to determine that the warning message is wrong.

Reply:
Perfect. You've worded that better than I could have. Thanks!